layouts/_default/events.html

```diff
@@ -51,68 +68,90 @@ <h2 class="text-2xl font-extrabold">
+            "heightRatio" 3
+            "widths" (slice 128 256 384 512 1280)
+            "figClass" "mt-2"
+            "sizeHint" "(min-width: 850px) 328px, 90vw"
+            "gravity" (index $params "image-gravity")
+          }}
+          {{ $eyebrow := $params.eventDate }}
+          {{ $rubricLink := $params.link }}
+          {{ $thumbClass := $params.thumbClass }}
+          {{ partial "tw/thumb-item.html" (dict
+            "wrapperClass" "flex flex-wrap gap-x-9"
+            "thumbClass" "w-full lg:max-w-md xl:max-w-lg"
+            "hedSize" `text-xl md:text-xl lg:text-2xl font-semibold`
+            "dekSize" "sm:text-base sm:leading-relaxed md:text-lg lg:text-xl"
+            "imgOpts" $imgOpts
+            "rubricLink" $rubricLink
+            "eyebrow" $eyebrow
+            "link" $params.link
+            "hed" $event.LinkTitle
+            "byline" $params.byline
+            "showDate" false
+            "showByline" $params.showByline
+            "dek" $params.blurb
+            )
+          }}
+        </div>
       </div>
+
+      {{ $pages := $upcomingEvents | after 1 | first 4 }}
+      {{ range $i, $page := $pages }}
+        {{ $params := partialCached "helper/page-params" . .RelPermalink }}
+        {{ $imgOpts := dict
+          "filename" (.Param "image")
+          "credit" (.Param "image-credit")
+          "description" (.Param "image-description")
+          "link" $params.link
+          "widthRatio" 16
+          "heightRatio" 9
+          "widths" (slice 128 256 384 512 1280)
+          "figClass" "mt-2"
+          "sizeHint" "(min-width: 850px) 328px, 90vw"
+          "gravity" (.Param "image-gravity")
+        }}
+        <div
+          class="flex max-w-full md:w-2/12 md:max-w-[272px] md:flex-1 lg:w-1/6 xl:w-72"
+          data-ga-category="story-{{ add 2 $i }}"
+        >
+          {{ $eyebrow := $params.eventDate }}
+          {{ $rubricLink := $params.link }}
+          {{ $thumbClass := $params.thumbClass }}
+          {{ partial "tw/thumb-item.html" (dict
+            "wrapperClass" "flex flex-wrap gap-x-4"
+            "thumbClass" "w-2/5 md:w-full"
+            "imgOpts" $imgOpts
+            "rubricLink" $rubricLink
+            "eyebrow" $eyebrow
+            "link" $params.link
+            "hed" .LinkTitle
+            "byline" $params.byline
+            "showDate" false
+            )
+          }}
+        </div>
+      {{ end }}
+
     </section>
   {{ end }}
 {{ end }}
 
-{{ define "upcoming-event-lg" }}
-  <div class="mx-auto max-w-screen-lg">
-    {{ $feat := index . 0 }}
-    {{ $params := partialCached "helper/page-params" $feat $feat.RelPermalink }}
-    {{ $subitems := slice }}
-    {{ range after 1 . }}
-      {{ $params := partialCached "helper/page-params" . .RelPermalink }}
-      {{ $subitems = $subitems | append (dict
-        "link" $params.link
-        "title" .Title
-        )
-      }}
-    {{ end }}
-    {{ $imgOpts := dict
-      "filename" $params.image
```

Comment:
This was a bug because the $params object doesn't have image on it. It's on the page.Params.

Reply:
I'm not sure I understand the different use-cases for `$params` vs just `$event`

Reply:
$params is the `partialCached "helper/page-params"` object that does some logic for us of calculating the pretty event date, author name, eyebrow links, etc. It exists so we have one place that does the fallback logic for that stuff when a parameter is missing or needs to fallback to different stuff. $event.Params is the native Hugo params dictionary.